jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -65,14 +66,19 @@ class S3TreeTable(
         }
     }
 
-    private val doubleClickListener = object : MouseAdapter() {
-        override fun mouseClicked(e: MouseEvent) {
-            if (e.clickCount < 2 || e.button != MouseEvent.BUTTON1) {
-                return
-            }
-            val row = rowAtPoint(e.point).takeIf { it >= 0 } ?: return
-            handleOpeningFile(row)
-            handleLoadingMore(row)
+    private val openFileListener = object : DoubleClickListener() {
+        override fun onDoubleClick(e: MouseEvent?): Boolean {
+            e ?: return false
+            val row = rowAtPoint(e.point).takeIf { it >= 0 } ?: return false
+            return handleOpeningFile(row)
+        }
+    }
+
+    private val loadMoreListener = object : DoubleClickListener() {
+        override fun onDoubleClick(e: MouseEvent?): Boolean {
```

Comment:
You can safely remove the null check here - `e` can never be null : https://github.com/JetBrains/intellij-community/blob/master/platform/platform-api/src/com/intellij/ui/DoubleClickListener.java#L30

Reply:
Yep but ondoubleclick is not marked @NotNull for some reason:
`protected abstract boolean onDoubleClick(MouseEvent event);`